src/hook/useGrantPermission.ts

```diff
@@ -0,0 +1,58 @@
+import {useQueryClient, useQuery, useMutation, UseMutationOptions} from 'react-query';
+import * as AsyncStorage from 'src/service/AsyncStorage';
+import messaging, {FirebaseMessagingTypes} from '@react-native-firebase/messaging';
+import {Platform} from 'react-native';
+import {usePushTopics} from './usePushTopics';
+
+const PERMISSION_GRANTING_SCREEN_SHOWN = 'PERMISSION_GRANTING_SCREEN_SHOWN';
+
+export function useGrantPermission(options?: UseMutationOptions<boolean, unknown, void, unknown>) {
+  const {toggleTopic, topics} = usePushTopics();
+  const queryClient = useQueryClient();
+
+  const {data: granted, isLoading} = useQuery('permission_granted', async () => {
+    if (Platform.OS === 'android') {
+      return true;
+    }
+
+    return Platform.OS === 'ios' && permissionAllowed(await messaging().hasPermission());
+  });
+
+  const {mutate: grant} = useMutation(async () => {
+    const status = await messaging().requestPermission();
+    const allowed = permissionAllowed(status);
+
+    if (allowed) {
+      for (const topic of topics) {
+        await toggleTopic({id: topic.id, subscribe: true});
+      }
+    }
+
+    return allowed;
+  }, options);
+
+  const setPermissionGrantingToShown = async () => {
+    await AsyncStorage.setItem(PERMISSION_GRANTING_SCREEN_SHOWN, true);
+    queryClient.invalidateQueries('permission_granted');
+  };
+
+  return {grant, setPermissionGrantingToShown, granted, isLoading};
+}
+
+function permissionAllowed(status: FirebaseMessagingTypes.AuthorizationStatus) {
+  return status === messaging.AuthorizationStatus.AUTHORIZED || status === messaging.AuthorizationStatus.PROVISIONAL;
+}
+
+export function useShouldShowPushPermissionScreen() {
```

Comment:
We also need a way to check if user clicked on skip here. otherwise, it would pop up on every app start. 
<img width="313" alt="Screenshot 2021-07-19 at 13 05 31" src="https://user-images.githubusercontent.com/1168983/126150833-0325a979-54ed-4ce2-805c-ff8fc2ced0a1.png">

how would you think is the best way to do that?

Reply:
I would remove the Skip button.. Apps normally just prompt the user to allow or deny permissions, there is no option to skip

Reply:
yes that will fix everything. I will do that.